Yank/pipeline.py

```diff
@@ -1962,7 +1963,7 @@ def _get_number_box_waters(self, *args):
 # ==============================================================================
 
 def trailblaze_alchemical_protocol(thermodynamic_state, sampler_state, mcmc_move, state_parameters,
-                                   std_energy_threshold=0.5, threshold_tolerance=0.05,
+                                   checkpoint_path, std_energy_threshold=0.5, threshold_tolerance=0.05,
```

Comment:
Do we want to make checkpointing optional, or required via the Python API? This approach seems to make it required, but it seems like it would be simple to make it optional.

Reply:
I agree it would be useful to have it optional in the function (although exposing it in the YAML will be slightly more complicated). It's possible test are not passing also because you're defining a positional argument after a keyword one.